generators/app/templates/src/containers/Profile/profile.container.js

```diff
@@ -18,283 +18,311 @@ const defaulProfilePhoto = '/img/icon/empty-profile.svg';
  */
 
 export class Profile extends Component {
-  constructor (props) {
-    super (props);
+    constructor(props) {
+        super(props);
 
-    this.state = {
-      formMode: true,
-      isLoading: false,
-      photo: defaulProfilePhoto,
-      formFields: [],
-      newLinkNodes: [],
-      updatedFields: {}
+        this.state = {
+            formMode: true,
+            isLoading: false,
+            photo: defaulProfilePhoto,
+            formFields: [],
+            newLinkNodes: [],
+            updatedFields: {},
+        };
+    }
+    async componentDidMount() {
+        this.setState({ isLoading: true });
+        await this.fetchPhoto();
+        await this.fetchProfile();
+        this.setState({ isLoading: false });
+    }
+
+    changeFormMode = () => {
+        this.setState({ formMode: !this.state.formMode, updatedFields: {} });
+    };
+    setDefaultData = () => {
+        this.setState({ formFields: [...this.state.formFields] });
+    };
+    onCancel = () => {
+        this.changeFormMode();
+        this.setDefaultData();
     };
-  }
-  async componentDidMount () {
-    this.setState ({isLoading: true});
-    await this.fetchPhoto ();
-    await this.fetchProfile ();
-    this.setState ({isLoading: false});
-  }
 
```

Comment:
can you please document the isFormValid function so people know what it's doing? Especially those reduce calls.